app/Http/Controllers/Api/ServicesController.php

```
@@ -328,9 +328,14 @@ public function create_service(Request $request)
                 });
             }
             if ($oneClickService) {
+                $dockerComposeRaw = base64_decode($oneClickService);
+
+                // Validate for command injection BEFORE creating service
+                validateDockerComposeForInjection($dockerComposeRaw);
+
                 $service_payload = [
```

PHPMD (2.15.0)

The variable $service_payload is not named in camelCase. (undefined)

(CamelCaseVariableName)